src/gtkballs.c

```diff
@@ -129,21 +191,21 @@ int main(int argc, char **argv)
    bind_textdomain_codeset(PACKAGE, "UTF8");
 #endif /* ENABLE_NLS */
 
-   /* drop privileges after spawning child with extra privs */
-   if (score_setup() == -1)
-      return 1;
+   /* drop privileges */
    setregid(getgid(), getgid());
```

Comment:
I think this is not needed after your change. Before the program was supposed to be installed with the setuid bit, so that all users on the same machine could write to the score file.

No there's no need in setuid and dropping privileges.